pkg/config/config_test.go

```diff
@@ -525,18 +525,9 @@ func TestSetDevfileConfiguration(t *testing.T) {
 			currentDevfile: odoTestingUtil.GetTestDevfileObj(fs),
 			wantDevFile: parser.DevfileObj{
 				Ctx: devfileCtx.FakeContext(fs, parser.OutputDevfileYamlPath),
-				Data: &testingutil.TestDevfileData{
-					Commands: []devfilev1.Command{
-						{
-							Id: "devbuild",
-							CommandUnion: devfilev1.CommandUnion{
-								Exec: &devfilev1.ExecCommand{
-									WorkingDir: "/projects/nodejs-starter",
-								},
-							},
-						},
-					},
-					Components: []devfilev1.Component{
+				Data: func() data.DevfileData {
```

Comment:
@mik-dass @kadel I have made this change based on what seemed like the best option to me. If you have any suggestions, let me know. There are about 60 to 70 occurrences of this change across the unit tests and code.

Reply:
I feel this might make debugging tests harder since we are ignoring the errors. I feel the best approach would be to mock the `DevfileData` interface using gomock. That way we can remove a lot of the unnecessary struct values from functions like these but that probably should be done as a separate PR.

Reply:
About ignoring the error, I'm doing this because I don't feel like it's worth doing that in unit tests. But I don't think of it as a good idea so not defending it here.

About mocking `DevfileData`, I think something like that should even exist in `devfile/library` repo. Let me see or ask around.

Reply:
@dharmit @mik-dass 

1. yeah probably should not ignore err, makes it difficult to debug
2. to alleviate this PR, I can revert the changes to TestDevfileData and have a discussion around replacing it with gomock and whether this should stay in devfile/library

what do you guys think?

Reply:
@maysunfaisal I feel that would be great :+1:

Reply:
@mik-dass @dharmit I've not used gomock before for interfaces. So I can have a look and see the pain points that is being discussed here and what it means for devfile/library consumers.

In the mean time, I've put up a PR https://github.com/devfile/library/pull/72 to revert the test data deletion so that it doesn't affect odo as much. It has been approved and I will leave it up to @dharmit to merge(pls feel free) the PR and use the devfile/library latest in this PR if needed.

Reply:
@maysunfaisal I don't think that it's required to revert the changes just because we don't have mocking mechanism in place right now. My PR has gone ahead with making the changes that would make it compatible with changes introduced in https://github.com/devfile/library/pull/62.

Sure the changes I have made might not be the best approach, but we can improve that once mocking is available in `devfile/library` repo.

Merging https://github.com/devfile/library/pull/72 would be akin to taking a step backwards, and I'd like to avoid it unless others feel this is the right thing to do.

WDYT @mik-dass @kadel

Reply:
> EDIT: Not sure how I missed this comment - #4465 (comment). I still feel like it would be a step backwards. Instead of ignoring the error, I could add a t.Error. Wouldn't that be OK? @mik-dass @maysunfaisal

I don't mind the `t.Error()` approach for now. But I also OK with https://github.com/devfile/library/pull/72 as it gives us more flexibility.

Reply:
PTAL https://github.com/openshift/odo/pull/4465/commits/2ad804242dfdda5411326bad8c66a170bac86a43.
Still missing at a few places, but I tried my best. Mocking would help iron things out, I hope.

Reply:
Marking resolved since I've addressed the comments, IMO.